src/ansible_navigator/actions/_actions.py

```diff
@@ -95,7 +98,11 @@ def run_interactive(package: str, action: str, *args: Any, **_kwargs: Any) -> An
     """Call the given action's run"""
     action_cls = get(package, action)
     app, interaction = args
-    return action_cls(app.args).run(app=app, interaction=interaction)
+    try:
+        return action_cls(app.args).run(app=app, interaction=interaction)
+    except AttributeError:
+        logger.error("Subcommand '%s' does not support mode interactive", action)
+        return action_cls(app.args).no_interactive_mode(app=app, interaction=interaction)
```

Comment:
In this place, try/except approach may be shadowing a few unobvious problem origins. This exception handler will be chosen in the following cases:
1) If `app` does not have `args` attribute
2) If internals of `action_cls()` (or *anything* down the call stack) raise `AttributeError`
3) If the object returned by `action_cls()` does not have a `run` attribute
4) If the `run` attribute exists, is callable but internally something raises `AttributeError`.

It wouldn't be nice if half of instructions/mutations inside of `run()`, error out, only to cause `no_interactive_mode()` to potentially repeat some of them, right? This is why in my original suggestion, I tried to minimize what's expected to be caught in the try/except block. This is also why https://github.com/ansible/ansible-language-server/issues/16 is a thing — it is important to scope the instructions within try/except to what you really expect and let any other exceptions bubble up naturally.

While the scenarios I described may not be happening right now, it is important to understand how dangerous this may become in the future and how hard it would be to debug such cases if they happen.

This is why I'd like to re-iterate the original proposal:
1) Initialize the instance outside of try/except
2) Extract the callable within try/except so that it's the only thing that is processed by this construct
3) Call it outside of try/except

Reply:
thanks for the explanation, I hadn't considered how deeply the except would go.

btw: https://github.com/ansible/ansible-navigator/pull/741